src/ocl_kernel.cpp

```diff
@@ -61,6 +64,12 @@ void OCLKernelInvoke::run() const
 
     MIOPEN_HANDLE_LOCK
 
+    const char* const arch = miopen::GetStringEnv(MIOPEN_DEVICE_ARCH{});
+    if(arch != nullptr && strlen(arch) > 0)
+    {
+        MIOPEN_THROW("Escaping launching kernel");
```

Comment:
Not an error when arch matches machine. More descriptive error would be better. eg compile arch does not match, escaping launching kernel / MIOPEN_DEVICE_ARCH was used, escaping launching kernel.